### PR TITLE
Feat/sortable questions

### PR DESCRIPTION
**Pull request by z-rayc**

### CODE REVIEW

File: internal/web_server/web/handlers/api/admin_api.go [hunk 1]
```diff
@@ -449,6 +450,34 @@ func (aah *AdminApiHandler) deleteArticle(c echo.Context) error {
...
+	// Rearrange the questions
+	err = questions.RearrangeQuestions(aah.sharedData.DB, c.Request().Context(), quizID, questionsList)
+	if err != nil {
+		if err == questions.ErrNonSequentialQuestions {
+			return utils.Render(c, http.StatusBadRequest, components.ErrorText("error-question-list", "Spørsmålene må ha en sekvensiell rekkefølge"))
+		}
+		return err
+	}
+
+	return c.NoContent(http.StatusOK)
```
Comment (nokacper24) [inline internal/web_server/web/handlers/api/admin_api.go:478]: i see you're doing no content, not replacing the table. This is probably okay, unless the list of questions on the page can go out of sync with the database order?

File: internal/web_server/web/views/pages/dashboard_pages/edit_quiz.templ [hunk 2]
```diff
@@ -253,23 +276,52 @@ script questionList() {
...
+					headers: {
+						"Content-Type": "application/json",
+						"HX-Swap": "outerHTML",
+						"HX-Target": "#question-list",
+					},
+					body: JSON.stringify(idMap),
+				}).then((response) => {
+					let errorText = document.querySelectorAll(".error-question-list")[0];
+
+					if (response.status >= 400) {
```
Comment (nokacper24) [inline internal/web_server/web/views/pages/dashboard_pages/edit_quiz.templ:310]: if there is an error, will the dragged item go back into its original place? or how does sortable work?
related to my other comment

**Review by nokacper24 — COMMENTED** [review]
generally looks ok i think, not braining too well atm so someone else look

File: internal/web_server/web/handlers/api/admin_api.go (hunk 1 above)
Comment (z-rayc) [reply internal/web_server/web/handlers/api/admin_api.go:478]: I send the whole list every time, so if one request in the middle isn't sent for some reason, it should update afterwards anyway. And if the last request happens to not be sent, it's not the end of the world. This is much simpler than swapping the table, which would probably add 50-100 more lines of code.

File: internal/web_server/web/views/pages/dashboard_pages/edit_quiz.templ (hunk 2 above)
Comment (z-rayc) [reply internal/web_server/web/views/pages/dashboard_pages/edit_quiz.templ:310]: No, it does not.

**Review by Molnes — APPROVED** [review]

**Pull request merged**